adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -1226,11 +1185,75 @@ protected final void onCommandReceived(
         commandContext.getCurrentSpan().log(items);
     }
 
-    private static void addRetainAnnotation(final MqttContext context, final Message downstreamMessage, final Span currentSpan) {
+    private static void addRetainAnnotation(final MqttContext context, final Message downstreamMessage,
+            final Span currentSpan) {
 
         if (context.message().isRetain()) {
             currentSpan.log("device wants to retain message");
             MessageHelper.addAnnotation(downstreamMessage, MessageHelper.ANNOTATION_X_OPT_RETAIN, Boolean.TRUE);
         }
     }
+
+    private Future<TenantObject> isAdapterEnabled(final TenantObject tenantConfig) {
+        if (tenantConfig.isAdapterEnabled(getTypeName())) {
+            LOG.debug("protocol adapter [{}] is enabled for tenant [{}]",
+                    getTypeName(), tenantConfig.getTenantId());
+            return Future.succeededFuture(tenantConfig);
+        } else {
+            LOG.debug("protocol adapter [{}] is disabled for tenant [{}]",
+                    getTypeName(), tenantConfig.getTenantId());
+            return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_FORBIDDEN,
+                    "adapter disabled for tenant"));
+        }
+    }
+
+    private Future<DeviceUser> authenticateCredentials(final DeviceCredentials credentials) {
+        final Future<DeviceUser> result = Future.future();
+        usernamePasswordAuthProvider.authenticate(credentials, result.completer());
+        return result;
+    }
+
+    private Future<Device> isDeviceRegistered(final Device authenticatedDevice, final DeviceCredentials credentials,
+            final Span currentSpan) {
+        currentSpan.log(String.format("device authenticated"));
+        LOG.debug("successfully authenticated device [tenant-id: {}, auth-id: {}, device-id: {}]",
+                authenticatedDevice.getTenantId(), credentials.getAuthId(),
+                authenticatedDevice.getDeviceId());
```

Comment:
the five lines above do not belong into this method IMHO. They should be moved to the *authenticateCredentials* method, don't you think?

Reply:
These debug statements already existed and I understood that the intention was to execute them once authentication succeeds. You are right and IMHO it doesn't belong here or `authenticateCredentials` method. I will change so that these statements are executed after successful authentication.